README.md

```diff
@@ -670,187 +660,31 @@ C.prototype.m = bound(
 
 ```js
-const MY_META = Symbol();
-const OTHER_META = Symbol();
-
-function myMeta(value, context) {
-  context.setMetadata(MY_META, 123);
-  context.setMetadata(OTHER_META, 123);
-}
-```
-
-All metadata defined under a particular symbol is gathered across the class and placed into a metadata object with the following interface:
-
-```ts
-interface Metadata {
-  constructor?: unknown;
-  public?: Record<string, unknown>;
-  private?: unknown[];
-}
-```
-
-Two of these objects may exist, one for static elements and the class itself, and one for non-static elements. These objects are then accessed via the `Symbol.metadata` property on the class, like so:
-
-```js
-target[Symbol.metadata][key];
-```
-
-Where `key` also must be a symbol. Static element metadata and class decorator metadata is accessed on the constructor directly, and non-static element metadata is accessed on the prototype.
-
-Given the following example decorator and class:
-
-```js
-const MY_META = Symbol();
-
-function myMeta(value, context) {
-  context.setMetadata(MY_META, 'metadata');
-}
-
-@myMeta
-class C {
-  @myMeta a = 123
-  @myMeta b() {}
-  @myMeta #c = 456;
-
-  @myMeta static x = 123;
-  @myMeta static y() {}
-  @myMeta static #z = 456;
-}
-```
-
-You would be able to access the metadata for this class like so:
-
-```js
-C.prototype[Symbol.metadata][MY_META];
-// {
-//   public: {
-//     a: 'metadata',
-//     b: 'metadata',
-//   },
-
-//   private: ['metadata']
-// }
-
-C[Symbol.metadata][MY_META];
-// {
-//   constructor: 'metadata',
-
-//   public: {
-//     x: 'metadata',
-//     y: 'metadata',
-//   },
-
-//   private: ['metadata']
-// }
-```
-
-Subsequent definitions to the same metadata key overwrite the existing value, if one exists. If decorators wish to instead build upon existing metadata, they can use the `getMetadata` method of the context object to get the current value and modify it.
-
-```js
-const VALIDATIONS = Symbol();
-
-function validateString(value, context) {
-  let validations = context.getMetadata(VALIDATIONS) ?? [];
-
-  validations.push((value) => typeof value === "string");
-
-  context.setMetadata(VALIDATIONS, validations);
-}
-
-function validateMaxLength(length) {
-  return (value, context) => {
-    let validations = context.getMetadata(VALIDATIONS) ?? [];
-
-    validations.push((value) => value.length < length);
-
-    context.setMetadata(VALIDATIONS, validations);
-  }
-}
-
-class C {
-  @validateString
-  @validateMaxLength(10)
-  foo = "hello!";
+function dec(value, context) {
+  context.addMetadata(123);
 }
-
-C.prototype[Symbol.metadata][VALIDATIONS].public.foo.length; // 2
 ```
 
-In addition, if two public class elements exist on the same class with the same name, then metadata defined on the second element will overwrite metadata defined on the element.
+Metadata values which are added this way are then exposed in an array on the `Symbol.metadata` property of the value being decorated.
```

Comment:
It looks like its not so much the "value being decorated", rather, the current class.

Reply:
This is meant to describe the paradigm, in the future it might refer to the function being decorated, etc. Methods are part of the class, so it made sense to me that the class is the overall “value” being decorated.

Reply:
While methods are part of the class, when it comes to method decorators they're also a distinct value on their own and are the value "being decorated"  - at least thats what the context is telling us.  I would think there's got to be a better way to word this to make that distinction.

Also, where is the line drawn?  Given future extensions, where would metadata for these decorators defined?

```javascript
class C {
    field: @dec function expression(){}
    method(@dec parameter){}
}
```

Reply:
Fields are also a distinct value, yet have no way of being represented other than the class itself. So this interpretation of a “value” is more consistent in that sense, all class element metadata ends up on the class itself. 

Parameter decorators do need to be considered, though they are out of scope for this proposal. I think it would make sense to add them to the methods/functions they are decorating. One possibility here would be to have method metadata be on the method, and then also be added to the class for convenience. 

My main concern with having method metadata solely be on the method is the complexity it would introduce for getters/setters. However, I suppose it would be the “simplest” solution, metadata ends up on the decorated value (or nearest parent, in the case of fields and parameters) in a simple array, no other complexities.